File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> out=new ArrayList<>();
        Stack<TreeNode> al=new Stack<>();
        if(root==null)return out;
        // al.push(root);
        TreeNode curr=root;
        while(!al.isEmpty() || curr!=null)
        {
            while(curr!=null)
            {
                al.push(curr);curr=curr.left;
            }
            curr=al.pop();
            out.add(curr.val);
            curr=curr.right;
        }
        return out;

    }
}

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        al=[]
        self.inorder(root,al)
        return al

    def inorder(self,root,al):
        if root is None:
            return 

        self.inorder(root.left,al)
        al.append(root.val)
        self.inorder(root.right,al)
